pkg/hubble/parser/threefour/parser.go

```diff
@@ -316,13 +316,22 @@ func (p *Parser) resolveEndpoint(ip net.IP, datapathSecurityIdentity uint32) *pb
 	if p.endpointGetter != nil {
 		if ep, ok := p.endpointGetter.GetEndpointInfo(ip); ok {
 			epIdentity := resolveIdentityConflict(ep.GetIdentity())
-			return &pb.Endpoint{
+			e := &pb.Endpoint{
 				ID:        uint32(ep.GetID()),
 				Identity:  epIdentity,
 				Namespace: ep.GetK8sNamespace(),
 				Labels:    sortAndFilterLabels(p.log, ep.GetLabels(), epIdentity),
 				PodName:   ep.GetK8sPodName(),
 			}
+			if ep.GetPod() != nil {
```

Comment:
nit: 
I know that the endpoint has a cached copy of the pod, but I would prefer for the `GetPod` method to be called once and saved to a variable in the if statement to be checked and reused.

Reply:
Do you mean add another pointer in the Parser struct and reuse that? GetPod just return a Pod pointer.